components/camel-azure/src/main/java/org/apache/camel/component/azure/common/AbstractConfiguration.java

```diff
@@ -48,4 +55,38 @@ public StorageCredentials getCredentials() {
     public void setCredentials(StorageCredentials credentials) {
         this.credentials = credentials;
     }
+
+    public String getCredentialsAccountKey() {
+        return credentialsAccountKey;
+    }
+
+    /**
+     * Set the storage account key used during authentication phase
+     */
+    public void setCredentialsAccountKey(String credentialsAccountKey) {
+        this.credentialsAccountKey = credentialsAccountKey;
+    }
+
+    public String getCredentialsAccountName() {
+        return credentialsAccountName;
+    }
+
+    /**
+     * Set the storage account name used during authentication phase
+     */
+    public void setCredentialsAccountName(String credentialsAccountName) {
+        this.credentialsAccountName = credentialsAccountName;
+    }
+
+    public  StorageCredentials getAccountCredentials() {
+        StorageCredentials creds = credentials;
+        //if  credentials is null, fallback to credentialsAccountKey and credentialsAccountName
+        if (creds == null) {
+            if (credentialsAccountKey != null && credentialsAccountName != null) {
```

Comment:
can we introduce another check if either `credentialsAccountKey` or `credentialsAccountName` is null and perhaps throw an error if `creds` and one of the `credentialsAccountKey` or `credentialsAccountName` are `null`?

Reply:
@omarsmak those checks are covered by BlobServiceComponent.checkCredentials and QueueServiceComponent.checkCredentials. Added unit tests to check those scenarios.

Reply:
Perfect! Thanks a lot